hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/tables/datanodesTable.tsx

```diff
@@ -0,0 +1,315 @@
+          <DecommissionSummary uuid={uuid} /> : <span>{uuid}</span>
+      );
+    }
+  },
+  {
+    title: 'Storage Capacity',
+    dataIndex: 'storageUsed',
+    key: 'storageUsed',
+    sorter: (a: Datanode, b: Datanode) => a.storageRemaining - b.storageRemaining,
+    render: (_: string, record: Datanode) => (
+      <StorageBar
+        strokeWidth={6}
+        capacity={record.storageTotal}
+        used={record.storageUsed}
+        remaining={record.storageRemaining}
+        committed={record.storageCommitted} />
+    )
+  },
+  {
+    title: 'Last Heartbeat',
+    dataIndex: 'lastHeartbeat',
+    key: 'lastHeartbeat',
+    sorter: (a: Datanode, b: Datanode) => moment(a.lastHeartbeat).unix() - moment(b.lastHeartbeat).unix(),
+    render: (heartbeat: number) => {
+      return heartbeat > 0 ? getTimeDiffFromTimestamp(heartbeat) : 'NA';
+    }
+  },
+  {
+    title: 'Pipeline ID(s)',
+    dataIndex: 'pipelines',
+    key: 'pipelines',
+    render: (pipelines: Pipeline[], record: Datanode) => {
+      const renderPipelineIds = (pipelineIds: Pipeline[]) => {
+        return pipelineIds?.map((pipeline: any, index: any) => (
+          <div key={index} className='pipeline-container-v2'>
+            <ReplicationIcon
+              replicationFactor={pipeline.replicationFactor}
+              replicationType={pipeline.replicationType}
+              leaderNode={pipeline.leaderNode}
+              isLeader={pipeline.leaderNode === record.hostname} />
+            {pipeline.pipelineID}
+          </div >
+        ))
+      }
+
+      return (
+        <Popover
+          content={
+            renderPipelineIds(pipelines)
+          }
+          title="Related Pipelines"
+          placement="bottomLeft"
+          trigger="hover">
+          <strong>{pipelines.length}</strong> pipelines
+        </Popover>
+      );
+    }
+  },
+  {
+    title: () => (
+      <span style={headerIconStyles} >
+        Leader Count
+        <Tooltip
+          title='The number of Ratis Pipelines in which the given datanode is elected as a leader.' >
+          <InfoCircleOutlined style={{ paddingLeft: '4px' }} />
+        </Tooltip>
+      </span>
+    ),
+    dataIndex: 'leaderCount',
+    key: 'leaderCount',
+    sorter: (a: Datanode, b: Datanode) => a.leaderCount - b.leaderCount
+  },
+  {
+    title: 'Containers',
+    dataIndex: 'containers',
+    key: 'containers',
+    sorter: (a: Datanode, b: Datanode) => a.containers - b.containers
+  },
+  {
+    title: () => (
+      <span style={headerIconStyles}>
+        Open Container
+        <Tooltip title='The number of open containers per pipeline.'>
+          <InfoCircleOutlined style={{ paddingLeft: '4px' }} />
+        </Tooltip>
+      </span>
+    ),
+    dataIndex: 'openContainers',
+    key: 'openContainers',
+    sorter: (a: Datanode, b: Datanode) => a.openContainers - b.openContainers
+  },
+  {
+    title: 'Version',
```

Comment:
We will also need to add search functionality for the Version column in the new UI.